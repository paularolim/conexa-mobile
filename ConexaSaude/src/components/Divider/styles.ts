import { View } from 'react-native';
import styled from 'styled-components/native';

export const Container = styled(View)`
  flex-direction: row;
  gap: 10px;
`;

export const Item = styled(View)<{ primary?: boolean }>`
  width: 80px;
  height: 5px;
  background-color: ${({ theme, primary }) =>
    primary ? theme.colors.pink : theme.colors.grayLight};
  border-radius: 2px;
`;
