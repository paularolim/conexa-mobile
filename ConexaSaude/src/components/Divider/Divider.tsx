import { Container, Item } from './styles';

export function Divider() {
  return (
    <Container>
      <Item primary />
      <Item />
    </Container>
  );
}
