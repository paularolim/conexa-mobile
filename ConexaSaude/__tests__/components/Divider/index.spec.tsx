import renderer from 'react-test-renderer';

import { Divider } from '@components/Divider';

import { Wrapper } from '../../Wrapper';

describe('Divider component', () => {
  it('should render default component correctly', () => {
    const tree = renderer
      .create(
        <Wrapper>
          <Divider />
        </Wrapper>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
