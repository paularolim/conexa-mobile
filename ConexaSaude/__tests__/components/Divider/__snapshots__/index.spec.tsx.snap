// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider component should render default component correctly 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
        "gap": 10,
      },
    ]
  }
>
  <View
    primary={true}
    style={
      [
        {
          "backgroundColor": "#F28080",
          "borderRadius": 2,
          "height": 5,
          "width": 80,
        },
      ]
    }
  />
  <View
    style={
      [
        {
          "backgroundColor": "#C2CFE0",
          "borderRadius": 2,
          "height": 5,
          "width": 80,
        },
      ]
    }
  />
</View>
`;
